### ISSUE TITLE
Method to identify type of plot with another plugin

### ISSUE THREAD

**Issue by keenerb** [echurchill/CityWorld#16]
I'd like some way to be able to identify, with another plugin, which type of plot a player is currently standing in.

Any suggestions?  I know with a normal cityworld you can use biome to check, but obviously that doesn't work in one of your desert/snow worlds.

What I ended up doing in an old fork of your plugin was hide specific types of wool in the bedrock layer; each chunk's 1,0,1 block was a specific color of wool capped with another bedrock block, and I just checked the damage value for the 1,0,1 block every time I needed to know what "type" of cityworld chunk I was in.

If you can think of something better or more elegant than this I'd really like to know.  I have a chat-based minimap that uses colored characters to indicate nearby plot types, it's quite good and I tie in other indicators as well like the chunk's radioactivity and mob count and etc.

**Comment by echurchill** [comment]
I plan to look at this weekend. Sorry for the delay

**Comment by keenerb** [comment]
Ideally this would be in a 1.7.10 compatible build, for what it's worth.  If that's even possible now.

**Comment by echurchill** [comment]
Sorry, I have been off line due to an injury. I hope to get back to this next weekend.

**Comment by keenerb** [comment]
No worries.  Get to it when (or IF) you have the time, it's nothing earth-shattering.  I could just do one or two neat things wtih the feature...

**Comment by keenerb** [comment]
I finally found a way to do this, for my very particular circumstances anyway, using the Skript plugin.

I set the underlying bedrock layer to 2 blocks rather than 1, and essentially turned the bottom layer into a bitmapped data set.

I added a bit of code to the very end of each plot's GenerateActualBlocks method that sets block 1, 1, 1 in a chunk to a colored block where the data value stores which plat type (Astral/Floating/Nature/Urban/etc.) the chunk belongs to:

chunk.setBlockTypeAndColor(1, 1, 1, Material.STAINED_CLAY, DyeColor.GRAY);

Using wool/clay/glass gives me 48 possible values for each block in the bottom layer, which means I can technically store 768 bits of data this way.

Block 1, 1, 2 is set to a colored block where data value 1 = farm plot, 2 = barn plot, 3 = road plot, etc. etc.

Block 1, 1, 3 is a colored block that contains the generator type that was in use when the chunk was created.

And so on and so forth.  With Skript I have a small routine that I can use to identify particular information about a chunk and use that in my script logic for a number of purposes, like equipping zombies with hoes/axes/leather helmets (hard hats) if they spawn in farmland/forests/unfinished building lots.
